### ISSUE TITLE
firebase-messaging update payload 

### ISSUE THREAD

**Issue by MSkred** [nuxt-community/firebase-module#154]
Hi all, I know the documentation said 

> we expect the notification format payload be named notification and can contain the following config

But it's possible to update the key name ( _exemple: notification ---> data_) 

Second question I see in the `firebase-messaging-sw.js` file this part of code

```
messaging.setBackgroundMessageHandler(function(payload) {
  console.info("SW received the message: ", payload);
  const notification = payload.notification;

  const notificationTitle = notification.title;
  const notificationOptions = {
    body: notification.body,
    icon: notification.image,
    vibrate: notification.vibrate || [200, 100, 200, 100, 200, 100, 200],
    actions: [
      // First item is always taken as click action (see comment below)
      {
        title: "Visit",
        action: notification.clickPath
      }
    ]
  };
  return self.registration.showNotification(
    notificationTitle,
    notificationOptions
  )
})
```

So, how can I add badge or icon in my notification it's possible ?

**Comment by lupas** [comment]
## Answers
@MSkred 
Thanks for the issue. This feature is currently still **experimental** and therefore we haven't really implemented various option to configure the service worker yet. So what you are asking is currently **not possible**.

If you want want customize the service-worker to your needs, you have two options:

1. Copy+Paste our service worker code, change it to your needs and implement it yourself - **FOR NOW** until someone found time to build more features into this functionality

2. Submit a PR that adds this functionality, I'm happy to have a look at it. :)

I'm a bit busy these days but maybe someone has time to look into it, shouldn't take too much time.

## How I think we can improve it

> But it's possible to update the key name ( exemple: notification ---> data)

We currently do this:
```js
const notification = payload.notification;
```

I guess we could do this...
```js
// Add option notificationKey to module-options
const notification = payload[options.notificationKey];
```

> So, how can I add badge or icon in my notification it's possible ?

I guess we could also pass all notificationOptions directly from the payload...

I'll keep the issue open until this is fixed, it's something we should have a look at soon.

**Comment by lupas** [comment]
Fixed with: https://github.com/nuxt-community/firebase-module/pull/219

We are now expected the use of the messaging HTTP v1 API as described [here](https://firebase.google.com/docs/cloud-messaging/js/first-message#add_web_push_properties_to_a_notification_payload) which will make life easier for multi-device support in general.

Instead of introducing a notificationKey element, we expect the payload to be in HTTP v1 API format.

Action listeners can be setup via the new `actions` property in the messaging options.

Will be released today with v6.0.0
